span.go

```diff
@@ -155,7 +155,25 @@ func (s *spanImpl) Tracer() opentracing.Tracer {
 }
 
 func (s *spanImpl) Context() opentracing.SpanContext {
-	return s.raw.SpanContext
+	return s.raw.Context
+}
+
+func (s *spanImpl) SetBaggageItem(key, val string) opentracing.Span {
+	s.onBaggage(key, val)
+	if s.trim() {
+		return s
+	}
+
+	s.Lock()
+	defer s.Unlock()
+	s.raw.Context = s.raw.Context.WithBaggageItem(key, val)
```

Comment:
I tend to agree.

BTW, the `basictracer-go` struct organization is probably wrong right now... Let me see if I can quickly adjust it this AM. My thinking is that we can get the `sync.Pool` back on by default and with no caveats about calling `Span.Context()` after `Span.Finish()`. (Calling other methods after `Span.Finish()` would be bad, though)